src/control/security/config_test.go

```diff
@@ -0,0 +1,220 @@
+		CertificateConfig: CertificateConfig{
+			CARootPath:      "testdata/certs/daosCA.crt",
+			CertificatePath: "testdata/certs/server.crt",
+			PrivateKeyPath:  "testdata/certs/server.key",
+			tlsKeypair:      nil,
+			caPool:          nil,
+		},
+	}
+}
+
+func AgentTC() *TransportConfig {
+	return &TransportConfig{
+		AllowInsecure: false,
+		CertificateConfig: CertificateConfig{
+			CARootPath:      "testdata/certs/daosCA.crt",
+			CertificatePath: "testdata/certs/agent.crt",
+			PrivateKeyPath:  "testdata/certs/agent.key",
+			tlsKeypair:      nil,
+			caPool:          nil,
+		},
+	}
+}
+
+func SetupTCFilePerms(t *testing.T, conf *TransportConfig) {
+	if err := os.Chmod(conf.CARootPath, SafeCertPerm); err != nil {
+		t.Fatal(err)
+	}
+	if err := os.Chmod(conf.CertificatePath, SafeCertPerm); err != nil {
+		t.Fatal(err)
+	}
+	if err := os.Chmod(conf.PrivateKeyPath, SafeKeyPerm); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func ValidateInsecure(t *testing.T, c *TransportConfig, err error) {
+	if err != nil {
+		t.Fatal(err)
+	}
+	if c.tlsKeypair != nil || c.caPool != nil {
+		t.Fatal("insecure config loaded certs")
+	}
+}
+
+func ValidateGood(t *testing.T, c *TransportConfig, err error) {
+	if err != nil {
+		t.Fatal(err)
+	}
+	if c.tlsKeypair == nil || c.caPool == nil {
+		t.Fatal("certs did not load yet returned no error")
+	}
+}
+
+func ValidateBad(t *testing.T, c *TransportConfig, err error) {
+	if err == nil {
+		t.Fatal(err)
+	}
+}
+
+func ValidateNil(t *testing.T, c *TransportConfig, err error) {
+	if err != nil &&
+		strings.Compare(err.Error(), "nil TransportConfig") != 0 {
+		t.Fatal(err)
+	}
+}
+
+func TestPreLoadCertData(t *testing.T) {
+	insecureTC := InsecureTC()
+	serverTC := ServerTC()
+	badTC := BadTC()
+
+	// Setup permissions for tests below.
+	SetupTCFilePerms(t, serverTC)
+	SetupTCFilePerms(t, badTC)
+
+	testCases := []struct {
+		testname string
+		config   *TransportConfig
+		Validate func(t *testing.T, c *TransportConfig, err error)
+	}{
+		{"InsecureTC", insecureTC, ValidateInsecure},
+		{"GoodTC", serverTC, ValidateGood},
+		{"BadTC", badTC, ValidateBad},
+		{"NilTC", nil, ValidateNil},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.testname, func(t *testing.T) {
+			err := tc.config.PreLoadCertData()
+			tc.Validate(t, tc.config, err)
+		})
+	}
+}
+
+func TestReloadCertData(t *testing.T) {
+	serverTC := ServerTC()
+	agentTC := AgentTC()
+	testTC := serverTC
+
+	SetupTCFilePerms(t, serverTC)
+	SetupTCFilePerms(t, agentTC)
+
+	err := testTC.PreLoadCertData()
+	if err != nil {
+		t.Fatal(err)
+	}
+	beforeCert := testTC.tlsKeypair.Certificate[0]
+
+	testTC.CertificatePath = agentTC.CertificatePath
+	testTC.PrivateKeyPath = agentTC.PrivateKeyPath
+
+	err = testTC.ReloadCertData()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	afterCert := testTC.tlsKeypair.Certificate[0]
+
+	if bytes.Compare(beforeCert, afterCert) == 0 {
+		t.Fatal("cert before and after reload is the same")
+	}
+}
+
+func TestPrivateKey(t *testing.T) {
```

Comment:
This and TestPublicKey both feel like they should either use the table driven format, or be split into two functions, since they each represent two test cases.

Reply:
reworked as table tests.